src/game_events/action_wml.cpp

```diff
@@ -894,8 +894,16 @@ WML_HANDLER_FUNCTION(unit,, cfg)
 		.allow_rename_side(true)
 		.allow_show(true);
 
-	uc.add_unit(parsed_cfg, &cfg);
+	try
+	{
+		uc.add_unit(parsed_cfg, &cfg);
+	}
+	catch(const unit_type::error& e)
+	{
+		ERR_WML << "Error occured inside [unit] WML.";
```

Comment:
```suggestion
		ERR_WML << "Error occured inside [unit]: " << e.what();
```

Reply:
if the exception is going to be rethrown is it necessary to also output it?

Reply:
What I mean is that it is going to be in the logs anyway.

Reply:
If we're bothering to output anything here then we might as well output the actual error message IMO.